### PR TITLE
Additional logging fixes

### PR DESCRIPTION
**Pull request by JonathanDCohen**
* Only logs from child processes in debug mode
* Logs exit codes or killing signals in prod mode
* Quiets BReactor logs
* Log disconnects and successful exit codes in `console.log` and error exits in `console.error`

### CODE REVIEW

File: src/electron/process_manager.ts [hunk 1]
```diff
@@ -404,13 +396,33 @@ class Tun2socks extends ChildProcessHelper {
...
+      args.push('--channel-loglevel', 'BReactor', 'info');
+    }
 
     this.launch(args);
   }
 }
+
+function logExit(processName: string, exitCode?: number, signal?: string) {
+  const prefix = `[EXIT - ${processName}]: `;
+  const exitReason = exitCode ?? signal;
```
Comment (fortuna) [inline src/electron/process_manager.ts:416]: Remove `exitReason`. It's not needed.
This would be clearer:
```js
if (exitCode !== null) {

} else if if (signal !== null) {

}
```

File: src/electron/process_manager.ts [hunk 2]
```diff
@@ -404,13 +396,33 @@ class Tun2socks extends ChildProcessHelper {
     args.push('--netif-ipaddr', TUN2SOCKS_VIRTUAL_ROUTER_IP);
     args.push('--netif-netmask', TUN2SOCKS_VIRTUAL_ROUTER_NETMASK);
     args.push('--socks-server-addr', `${this.proxyAddress}:${this.proxyPort}`);
-    args.push('--loglevel', this.isInDebugMode ? 'debug' : 'error');
     args.push('--transparent-dns');
     if (isUdpEnabled) {
       args.push('--socks5-udp');
       args.push('--udp-relay-addr', `${this.proxyAddress}:${this.proxyPort}`);
     }
+    args.push('--loglevel', this.isInDebugMode ? 'debug' : 'error');
```
Comment (fortuna) [inline src/electron/process_manager.ts:404]: tun2socks is quite verbose too, but I don't really see anything useful in the debug logs.
Let's just set the level to `info` and remove the part below.

**Review by fortuna — CHANGES_REQUESTED** [review]

File: src/electron/process_manager.ts (hunk 1 above)
Comment (JonathanDCohen) [reply src/electron/process_manager.ts:416]: `exitReason` covers both null and undefined

File: src/electron/process_manager.ts (hunk 2 above)
Comment (JonathanDCohen) [reply src/electron/process_manager.ts]: Done in acee18b9a7e13e0554c4e5301128300ba8734d8e

File: src/electron/process_manager.ts (hunk 1 above)
Comment (fortuna) [reply src/electron/process_manager.ts:412]: It's harder to ensure it's correct.
What if signal == exitCode?
What if exitCode == null and signal == 0?

Let's be more explicit and test for null, which aligns well with the API documentation.

Comment (JonathanDCohen) [reply src/electron/process_manager.ts:412]: Done in 8268a9684b86840a8c0b77b436e0fc897addf7c4

Comment (JonathanDCohen) [reply src/electron/process_manager.ts]: Done in 92b15d7408ec15d5112140945ecf0bf5c8b57bb5

**Review by fortuna — APPROVED** [review]

**Review by fortuna — COMMENTED** [review]
Thanks for the fix! This will be very helpful in debugging desktop.
